### PR TITLE
Make pub all fields in Client struct

### LINKED ISSUE #165: Could you make Client's client id as pub?
Since Twitter's OAuth2 requires the client ID in the parameter as well, even though the urlencoded client ID is already included in the request body ( https://github.com/ramosbugs/oauth2-rs/blob/main/src/lib.rs#L1935 ), I would like Client struct to exposes the client_id parameter as pub or adding pub function.

I made little workaround to use this great library on twitter oauth2 as like
```Rust
// Get an auth token
let token = oauth_client
    .exchange_code(code)
    .add_extra_param("client_id", env::var("TWITTER_CLIENT_ID").context("Missing TWITTER_CLIENT_ID!").unwrap()) // Here!
    .set_pkce_verifier(session.get("pkce_code_verifier").unwrap())
    .request_async(async_http_client)
    .await
    .unwrap();
```

### PR DESCRIPTION
**Pull request by tomocrafter**
closes #165

### CODE REVIEW

**Comment by ramosbugs** [comment]
thanks for the PR! mind adding getter functions rather than directly exposing the fields? I think the main difference is that making the fields `pub` would allow for mutation, but I think an immutable interface is what we want here.

e.g.
```rust
pub fn client_id(&self) -> &ClientId {
  &self.client_id
}
```

**Comment by tomocrafter** [comment]
Should I put the details of the return value of the function in the doc comment of each function?

File: src/lib.rs [hunk 1]
```diff
@@ -965,6 +965,42 @@ where
             _phantom: PhantomData,
         })
     }
+
+    pub fn client_id(&self) -> &ClientId {
+        &self.client_id
+    }
+
+    pub fn client_secret(&self) -> Option<&ClientSecret> {
```
Comment (ramosbugs) [inline src/lib.rs:973]: let's leave out the client secret since it's safer for secrets to be write-only

**Review by ramosbugs — COMMENTED** [review]
> Should I put the details of the return value of the function in the doc comment of each function?

I don't have a strong opinion, but clippy is sad at the moment and it's probably better to put a short doc comment rather than disable the lint. Feel free to copy/paste the description from the setter and/or factory method

thanks for taking care of this!

**Pull request merged**